exporter/skywalkingexporter/metricrecord_to_metricdata.go

```diff
@@ -0,0 +1,234 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package skywalkingexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/skywalkingexporter"
+
+import (
+	"strconv"
+
+	"go.opentelemetry.io/collector/model/pdata"
+	metricpb "skywalking.apache.org/repo/goapi/collect/language/agent/v3"
+)
+
+const (
+	defaultServiceInstance = "otel-collector-instance"
+)
+
+func resourceToMetricLabels(resource pdata.Resource) []*metricpb.Label {
+	labels := make([]*metricpb.Label, 0)
+	attrs := resource.Attributes()
+	attrs.Range(func(k string, v pdata.AttributeValue) bool {
+		labels = append(labels,
+			&metricpb.Label{
+				Name:  k,
+				Value: v.AsString(),
+			})
+		return true
+	})
+	return labels
+}
+
+func numberMetricsToData(name string, data pdata.NumberDataPointSlice, defaultLabels []*metricpb.Label) (metrics []*metricpb.MeterData) {
+	metrics = make([]*metricpb.MeterData, 0)
+	for i := 0; i < data.Len(); i++ {
+		dataPoint := data.At(i)
+		attributeMap := dataPoint.Attributes()
+		labels := make([]*metricpb.Label, 0)
+		attributeMap.Range(func(k string, v pdata.AttributeValue) bool {
+			labels = append(labels, &metricpb.Label{Name: k, Value: v.AsString()})
+			return true
+		})
+
+		for _, label := range defaultLabels {
+			labels = append(labels, &metricpb.Label{Name: label.Name, Value: label.Value})
+		}
+		meterData := &metricpb.MeterData{}
+		sv := &metricpb.MeterData_SingleValue{SingleValue: &metricpb.MeterSingleValue{}}
+		sv.SingleValue.Labels = labels
+		meterData.Timestamp = dataPoint.Timestamp().AsTime().UnixMilli()
+		sv.SingleValue.Name = name
+		switch dataPoint.Type() {
+		case pdata.MetricValueTypeInt:
+			sv.SingleValue.Value = float64(dataPoint.IntVal())
+		case pdata.MetricValueTypeDouble:
+			sv.SingleValue.Value = dataPoint.DoubleVal()
+		}
+		meterData.Metric = sv
+		meterData.Service = defaultServiceName
+		meterData.ServiceInstance = defaultServiceInstance
```

Comment:
You may be interested in https://github.com/open-telemetry/opentelemetry-specification/blob/da74730bf835010229a9612c281f052e26553218/specification/resource/semantic_conventions/README.md#service (just a FYI, I think this can be a separate PR)

Reply:
Updated